### PR TITLE
Request transfer function from Livre on connect to init GUI

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: apps/livreGUI/Controller.cpp [hunk 1]
```diff
@@ -149,8 +149,12 @@ struct Controller::Impl
...
-            BOOST_FOREACH( const URISubscriberPair& pair, _subscriberMap )
+            SubscriberMap subscribers;
+            {
+                ScopedLock lock( _subscriberMutex );
+                subscribers = _subscriberMap;
+            }
+            BOOST_FOREACH( const URISubscriberPair& pair, subscribers )
             {
                 zeq::Subscriber* subscriber = pair.second;
                 subscriber->receive( 100 );
```
Comment (bilgili) [inline apps/livreGUI/Controller.cpp]: If two threads access the same subscriber for registering events and mean while that subscriber is on receive block this copying may not help.

File: apps/livreGUI/tfEditor/TransferFunctionEditor.cpp [hunk 2]
```diff
@@ -66,6 +69,9 @@ TransferFunctionEditor::TransferFunctionEditor( livre::Controller& controller,
     connect( ui->btnConnect, SIGNAL( pressed()), this, SLOT( _connect( )));
     connect( ui->btnDisconnect, SIGNAL( pressed()), this, SLOT( _disconnect()));
 
+    connect( this, &TransferFunctionEditor::transferFunctionChanged,
+             this, &TransferFunctionEditor::_onTransferFunctionChanged );
```
Comment (bilgili) [inline apps/livreGUI/tfEditor/TransferFunctionEditor.cpp]: this does not need queuedConnection ?

Comment (tribal-tec) [inline apps/livreGUI/tfEditor/TransferFunctionEditor.cpp]: Imo Qt does this automatically if connection type is AUTO and signaler is in different thread than the receiver.

File: apps/livreGUI/Controller.cpp (hunk 1 above)
Comment (tribal-tec) [inline apps/livreGUI/Controller.cpp]: Don't fully understand, but is solves deadlocks when registering/deregistering subscribers while message are received.

**Pull request merged**